cognite/client/_api/annotations.py

```diff
@@ -183,18 +183,13 @@ def list(self, filter: AnnotationFilter | dict, limit: int | None = DEFAULT_LIMI
         Returns:
             AnnotationList: list of annotations
         """
-        assert_type(limit, "limit", [int], allow_none=False)
```

Comment:
Any reason for not just changing to allow_none=True?

Reply:
Yes, it would fail for `math.inf` which is valid. Also, the `_list` method handles this for us automatically.

Reply:
So this is the actual fix? Remove the custom handling of `limit` and just pass it to `_list`?

Reply:
Yup!